launcher/minecraft/mod/ModFolderModel.cpp

```diff
@@ -162,18 +162,14 @@ Task* ModFolderModel::createParseTask(Resource& resource)
 
 bool ModFolderModel::uninstallMod(const QString& filename, bool preserve_metadata)
 {
-    for(auto mod : allMods()){
-        if(mod->fileinfo().fileName() == filename){
-            auto index_dir = indexDir();
-            mod->destroy(index_dir, preserve_metadata);
+    QList<Mod *> mods = allMods();
+    auto mod_it = std::find_if(mods.cbegin(), mods.cend(), [&filename](const auto& mod) { return mod->fileinfo().fileName() == filename; });
+    if (mod_it == mods.cend()) return false;
 
-            update();
-
-            return true;
-        }
-    }
-
-    return false;
+    QDir index_dir = indexDir();
+    (*mod_it)->destroy(index_dir, preserve_metadata);
```

Comment:
There may need to be a deeper dive to get the codebase const-correct as `destroy` could take a const reference if the 3 functions deeper in its call stack took const references too.